### PR TITLE
Add documentation section on client authentication

### PR DESCRIPTION
**Pull request by faph**
## What do these changes do?

Adds a dedicated section on client authentication using the `auth` argument for HTTP Basic Auth. Also explains how to configure other authentication flows including periodic renewal of credentials.

## Are there changes in behavior for the user?

No. Just makes it easier for developers to access authentication `aiohttp` features. Avoids users asking questions like this: https://github.com/aio-libs/aiohttp/discussions/6908

## Related issue number

Issue #6915 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by faph** [comment]
@Dreamsorcerer I am proposing this documentation enhancements, since there's been at least 2 people who were wondering how to do this recently. Let me know if you have any feedback.

**Comment by faph** [comment]
@webknjaz @Dreamsorcerer Please authorize workflows when you have a minute.

File: docs/client_advanced.rst
Comment (Dreamsorcerer) [inline docs/client_advanced.rst]:
```suggestion
creating a :class:`ClientSession` instance (as a :class:`CIMultiDict` object).
```

File: docs/client_advanced.rst
Comment (Dreamsorcerer) [inline docs/client_advanced.rst]: I feel like this is implied from the previous sentence.

File: docs/client_advanced.rst
Comment (Dreamsorcerer) [inline docs/client_advanced.rst]: I'd be inclined to revert these. They make the English slightly awkward and would need to become a little more wordy to make a natural sounding sentence.

Comment (faph) [reply docs/client_advanced.rst]: Understood. I have put the arrow back, but still proposing to drop the double back tick literal value syntax if that's ok. I found that a little confusing since we're not referencing some system identifiers or variable names etc.

Arrow is surrounded by narrow non-breaking spaces to make sure http-arrow-https is kept together on a single line.

**Comment by faph** [comment]
@Dreamsorcerer @webknjaz Comments addressed, please authorize pipes and review.

**Comment by faph** [comment]
@Dreamsorcerer Unlikely these failing tests on py 3.11 are related to the docs updates. Are we able to merge with failing tests? Or how do we resolve this?

**Comment by Dreamsorcerer** [comment]
> @Dreamsorcerer Unlikely these failing tests on py 3.11 are related to the docs updates. Are we able to merge with failing tests? Or how do we resolve this?

They don't block merging, they are marked as experimental.

**Review by Dreamsorcerer — APPROVED** [review]
Looks fine to me. @webknjaz Happy with adding this?

**Comment by faph** [comment]
@webknjaz Nudge :)

**Comment by faph** [comment]
@Dreamsorcerer any other maintainers beside @webknjaz who could approve this?

**Comment by faph** [comment]
@asvetlov possibly?

**Comment by Dreamsorcerer** [comment]
It's not like your changes will appear anywhere until we make a new release. There's no rush, we'll get back to it.

**Comment by faph** [comment]
@Dreamsorcerer I appreciate that. However, to clear my head it would be nice if a simple doc change could be merged in a reasonable timescale. 

I really would like to get back to the question how we could/should actually enhance the auth experience. 

Just being able to close out PRs as soon as possible seems good practice. It also is encouraging for the community of contributors :)

**Comment by faph** [comment]
Thanks @Dreamsorcerer

**Pull request merged**
